libexec/rtld-elf/rtld_c18n.c

```diff
@@ -1375,7 +1382,8 @@ tramp_make_entry(const struct tramp_header *header)
 }
 
 void *
-tramp_intern(const Obj_Entry *reqobj, const struct tramp_data *data)
+tramp_intern(const Obj_Entry *reqobj, compart_id_t caller,
```

Comment:
Why not just pass the Plt_Entry instead of the Obj_Entry and use reqplt->compart_id everywhere?

Reply:
Because of the reqobj == NULL special case in `tramp_should_include` for "internal" calls from rtld out to symbols like ifunc resolvers, etc.  Those calls don't come from a Plt_Entry (and in the future when we start checking data GOT entries we won't have a Plt_Entry either, but will have a valid `caller`).

For `tramp_should_include` I could perhaps  replace the `reqobj == NULL` check with `caller == RTLD_COMPART_ID` (which I currently assert) which would let me remove the `reqobj` argument from `tramp_should_include` and `tramp_intern`.

I did find myself wishing we had a wrapper macro or some such for all the internal calls instead of having the NULL pointers everywhere.  Also, `tramp_intern` is not a great name for this function.

Reply:
Note that we will lose the reqobj->path string in the tramp_check_sig error message in the `plt == NULL` case, but that's probably ok.  That code did not handle `reqobj == NULL` before.